### PR TITLE
[8.x] Add DB command to drop into the database CLI

### PR DESCRIPTION
**Pull request by paras-malhotra**
## Motivation

I currently use many aliases to connect to my database on my Laravel apps. It would be so much easier if we could just run `php artisan db` or `php artisan db --database=mysql` to quickly drop into the database CLI! No more aliases to remember for every app and easily switch between connections for apps that have multiple connections. This is similar to what Rails has in the `rails db` command.

## Screenshots

I've tested this on mysql, pgsql and sqlite. I don't have SQL Server but I think it should run without any errors.

Screenshot 2020-11-20 at 5 22 44 PM

### CODE REVIEW

**Comment by driesvints** [comment]
Think you can omit that flag and just make it an optional argument: `php artisan db mysql`.

**Comment by paras-malhotra** [comment]
@driesvints makes sense, made the change! Also renamed the title of the argument to `connection` instead of `database` to make it more obvious.`

File: src/Illuminate/Database/Console/DbCommand.php [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+     * @param  array  $connection
+     * @return array
+     */
+    protected function getSqlsrvArgs($connection)
+    {
+        return $this->buildOptionalArguments([
+            'database'      => '-d '.$connection['database'],
+            'username'      => '-U '.$connection['username'],
+            'password'      => '-P '.$connection['password'],
+            'host'          => '-S tcp:'.$connection['host']
```
Comment (driesvints) [inline src/Illuminate/Database/Console/DbCommand.php:161]: Please don't indent these kind of blocks like this.

File: src/Illuminate/Database/Console/DbCommand.php [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+        return $connection;
+    }
+
+    /**
+     * Get the arguments for the database client command.
+     *
+     * @param  array  $connection
+     * @return array
+     */
+    public function getArgs($connection)
```
Comment (driesvints) [inline src/Illuminate/Database/Console/DbCommand.php:68]: You can add the `array` typehint to each argument that requires it in the methods here and below.

**Comment by paras-malhotra** [comment]
@driesvints done! 👍

**Review by driesvints — APPROVED** [review]

File: src/Illuminate/Foundation/Providers/ArtisanServiceProvider.php [hunk 3]
```diff
@@ -88,6 +89,7 @@ class ArtisanServiceProvider extends ServiceProvider implements DeferrableProvid
         'ClearResets' => 'command.auth.resets.clear',
         'ConfigCache' => 'command.config.cache',
         'ConfigClear' => 'command.config.clear',
+        'Db' => DbCommand::class,
```
Comment (dinhquochan) [inline src/Illuminate/Foundation/Providers/ArtisanServiceProvider.php:92]: Why is this different? may be `command.db`

Comment (paras-malhotra) [reply src/Illuminate/Foundation/Providers/ArtisanServiceProvider.php:92]: These are all going away in 9x. Refer https://github.com/laravel/framework/pull/34925

**Pull request merged**